app/shared/ui-foundation/src/commonMain/kotlin/ui/foundation/HorizontalScrollNavigator.kt

```diff
@@ -0,0 +1,213 @@
+    scrollLeftButton: @Composable (modifier: Modifier) -> Unit = {
+        HorizontalScrollNavigatorDefaults.ScrollLeftButton(it)
+    },
+    scrollRightButton: @Composable (modifier: Modifier) -> Unit = {
+        HorizontalScrollNavigatorDefaults.ScrollRightButton(it)
+    },
+    content: @Composable () -> Unit
+) {
+    Box(
+        modifier = modifier
+            .pointerInput(Unit) {
+                awaitPointerEventScope {
+                    while (true) {
+                        val event = awaitPointerEvent(PointerEventPass.Initial)
+                        event.changes.firstOrNull()?.let { pointerInputChange ->
+                            state.updateMousePointerPosition(pointerInputChange.position)
+                        }
+                    }
+                }
+            },
+    ) {
+        content()
+
+        Box(
+            modifier = Modifier
+                .align(Alignment.CenterStart)
+                .padding(start = HorizontalScrollNavigatorDefaults.ButtonMargin)
+                .onGloballyPositioned { state.updateLeftButtonCoordinate(layoutCoordinates = it) },
+        ) {
+            Crossfade(targetState = state.showLeftButton) { show ->
+                if (show) {
+                    scrollLeftButton(Modifier.clickable { state.scrollLeft() })
+                }
+            }
+        }
+        Box(
+            modifier = Modifier
+                .align(Alignment.CenterEnd)
+                .padding(end = HorizontalScrollNavigatorDefaults.ButtonMargin)
+                .onGloballyPositioned { state.updateRightButtonCoordinate(layoutCoordinates = it) },
+        ) {
+            Crossfade(targetState = state.showRightButton) { show ->
+                if (show) {
+                    scrollRightButton(Modifier.clickable { state.scrollRight() })
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Create a [HorizontalScrollNavigatorState] that can be used to navigate horizontally
+ *
+ * @param scrollableState the incoming scrollable state. Use this to detect
+ *      if the content can be scrolled, then finally determine the visibility of navigation button.
+ * @param onClickNavigation called when clicked navigation button.
+ *      `step` is the scroll step, positive means scroll forward.
+ * @see HorizontalScrollNavigator
+ */
+@Composable
+fun rememberHorizontalScrollNavigatorState(
+    scrollableState: ScrollableState,
+    scrollStep: Float = HorizontalScrollNavigatorDefaults.ScrollStep,
+    onClickNavigation: (step: Float) -> Unit = { },
+): HorizontalScrollNavigatorState {
+    val scope = rememberCoroutineScope()
+    return remember(scrollableState, scrollStep, scope) {
+        HorizontalScrollNavigatorState(scrollableState, scrollStep, onClickNavigation, scope)
+    }
+}
+
+class HorizontalScrollNavigatorState(
+    private val scrollableState: ScrollableState,
+    private val scrollStep: Float,
+    private val onClickNavigation: (step: Float) -> Unit,
```

Comment:
```suggestion
    private val onScroll: (step: Float) -> Unit,
```

Should also add documentation. What does `step` mean? Is it just `scrollStep`?